### PR TITLE
easyeffects: fix bug 'missing spectrum analyzer'

### PR DESCRIPTION
**Pull request by soupglasses**
Due to old version of `lsp-plugins` in nixpkgs, the delay plugin is incorrectly inserting silence in the plugin output instead of the processed audio. This has been fixed in `lsp-plugins-1.2.17` but is not in nixpkgs yet.

Added a patch to revert the merge which causes issues, but the correct fix longer term will be to properly package `lsp-plugins-1.2.17`. I tried to this, but this release has also added gstreamer support to the package, but it is trying to do all sorts of odd things by writing into gstreamer's path instead of its own due to that new support.

Related issue in easyeffects: https://github.com/wwmm/easyeffects/issues/3394

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by soupglasses** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>easyeffects</li>
    <li>easyeffects.debug</li>
    <li>gnomeExtensions.easyeffects-preset-selector</li>
  </ul>
</details>

**Comment by soupglasses** [comment]
CC maintainers: @getchoo

File: pkgs/by-name/ea/easyeffects/package.nix [hunk 1]
```diff
@@ -114,6 +114,11 @@ stdenv.mkDerivation rec {
       )
     '';
 
+  patches = [
+    # Remove when lsp-plugins is >= 1.2.17. (https://github.com/wwmm/easyeffects/issues/3394)
+    ./0001-Revert-Merge-pull-request-3340-from-violetmage-viole.patch
```
Comment (AndersonTorres) [inline pkgs/by-name/ea/easyeffects/package.nix:119]: Use fetchpatch

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Reword the commit message:

```
easyeffects: fix bug " missing spectrum analyzer"
```

File: pkgs/by-name/ea/easyeffects/package.nix (hunk 1 above)
Comment (soupglasses) [reply pkgs/by-name/ea/easyeffects/package.nix:119]: This is not an upstream patch and created by me, so I'm not sure where i would fetchpatch from.

Comment (getchoo) [reply pkgs/by-name/ea/easyeffects/package.nix:119]: Github doesn't provide a .patch equivalent for reverting commits; this should be fine

**Review by getchoo — APPROVED** [review]
Thanks for investigating all of this! I hadn't heard about this issue just yet, and it probably would've taken me a good while to figure out right now 😆

The only nitpick I really have is that something like "easyeffects: fix issues with older versions of `lsp-plugins`" would probably be a better commit message, but the current one is fine as well really

**Comment by soupglasses** [comment]
I kept it short and used the description to explain it to keep it under 72 characters long. I can change it if it's important to include that in the title as well.

**Comment by getchoo** [comment]
> I can change it if it's important to include that in the title as well.

Completely up to you. I think either gets the message across well enough :)

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 344971` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>easyeffects</li>
    <li>easyeffects.debug</li>
    <li>gnomeExtensions.easyeffects-preset-selector</li>
  </ul>
</details>

File: pkgs/by-name/ea/easyeffects/package.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/by-name/ea/easyeffects/package.nix:119]: As far as I remember, fetchpatch can revert a patch.

**Review by getchoo — APPROVED** [review]

**Pull request merged**